src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/IRequestContextAdapter.cs

```diff
@@ -0,0 +1,31 @@
+// <copyright file="IRequestContextAdapter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Contrib.Instrumentation.AWS.Implementation;
+internal interface IRequestContextAdapter
+{
+    bool HasMessageBody { get; }
+
+    bool HasOriginalRequest { get; }
```

Comment:
Probably this should be renamed to something like
```suggestion
    bool CanInject { get; }
```

if that is the purpose?

I am not sure what we need HasMessageBody  for.

Reply:
Applied along with other changes.